tests/unit/fs_mock_metadata_to_disk_buffer_test.cc

```diff
@@ -0,0 +1,398 @@
+
+    mock_metadata_to_disk_buffer buf(default_buff_size, default_alignment);
+    buf.init(0);
+    BOOST_REQUIRE_EQUAL(buf.bytes_left(), default_buff_size - checkpoint_size);
+    BOOST_REQUIRE_EQUAL(buf.append(ondisk_delete_inode {1}), APPENDED);
+
+    auto buf2 = buf.virtual_constructor(default_buff_size, default_alignment);
+    buf2->init(default_alignment);
+
+    BOOST_REQUIRE_EQUAL(created_buffers.size(), 1);
+    BOOST_CHECK(created_buffers[0] == buf2);
+    BOOST_REQUIRE_EQUAL(created_buffers[0]->actions.size(), 0);
+    buf2->append(ondisk_delete_inode {1});
+    BOOST_REQUIRE_EQUAL(created_buffers[0]->actions.size(), 1);
+
+    auto buf3 = buf2->virtual_constructor(default_buff_size, default_alignment);
+    buf3->init(default_alignment * 2);
+
+    BOOST_REQUIRE_EQUAL(created_buffers.size(), 2);
+    BOOST_CHECK(created_buffers[1] == buf3);
+    BOOST_REQUIRE_EQUAL(created_buffers[1]->actions.size(), 0);
+}
+
+// tests below check that actions add correct info to actions vector
+
+SEASTAR_THREAD_TEST_CASE(flush_action_test) {
+    mock_metadata_to_disk_buffer buf(default_buff_size, default_alignment);
+    buf.init(0);
+    auto dev_impl = make_shared<mock_block_device_impl>();
+    block_device dev(dev_impl);
+
+    buf.flush_to_disk(dev).get();
+
+    BOOST_CHECK(buf.is_type<flush_to_disk>(0));
+}
+
+SEASTAR_THREAD_TEST_CASE(next_metadata_cluster_test) {
+    mock_metadata_to_disk_buffer buf(default_buff_size, default_alignment);
+    buf.init(0);
+
+    ondisk_next_metadata_cluster next_metadata_cluster_op {6};
+    BOOST_REQUIRE_EQUAL(buf.append(next_metadata_cluster_op), APPENDED);
+
+    BOOST_REQUIRE_EQUAL(buf.bytes_left(), default_buff_size - checkpoint_size -
+            get_ondisk_entry_size(next_metadata_cluster_op));
+
+    BOOST_CHECK(buf.is_append_type<ondisk_next_metadata_cluster>(0));
+    auto &elem = buf.get_by_append_type<ondisk_next_metadata_cluster>(0);
+    BOOST_REQUIRE_EQUAL(elem.cluster_id, next_metadata_cluster_op.cluster_id);
+}
+
+SEASTAR_THREAD_TEST_CASE(create_inode_test) {
+    mock_metadata_to_disk_buffer buf(default_buff_size, default_alignment);
+    buf.init(0);
+
+    ondisk_create_inode create_inode_op {42, 1, {5, 2, 6, 8, 4}};
+    BOOST_REQUIRE_EQUAL(buf.append(create_inode_op), APPENDED);
+
+    BOOST_REQUIRE_EQUAL(buf.bytes_left(), default_buff_size - checkpoint_size -
+            get_ondisk_entry_size(create_inode_op));
+
+    BOOST_CHECK(buf.is_append_type<ondisk_create_inode>(0));
+    auto &elem = buf.get_by_append_type<ondisk_create_inode>(0);
+    BOOST_REQUIRE_EQUAL(elem.inode, create_inode_op.inode);
+    BOOST_REQUIRE_EQUAL(elem.is_directory, create_inode_op.is_directory);
+    BOOST_CHECK(std::memcmp(&elem.metadata, &create_inode_op.metadata, sizeof(elem.metadata)) == 0);
+}
+
+SEASTAR_THREAD_TEST_CASE(update_metadata_test) {
+    mock_metadata_to_disk_buffer buf(default_buff_size, default_alignment);
+    buf.init(0);
```

Comment:
This repeats everywhere, maybe make a function like `default_mock_buf` for that?